src/chainstate/burn/mod.rs

```diff
@@ -236,32 +265,21 @@ impl ConsensusHash {
 
     /// Get the previous consensus hashes that must be hashed to find
     /// the *next* consensus hash at a particular block.
-    /// The resulting list will include the consensus hash at block_height.
-    pub fn get_prev_consensus_hashes<A, K>(conn: &Connection, block_height: u64, first_block_height: u64) -> Result<Vec<ConsensusHash>, db_error>
-    where
-        A: Address,
-        K: PublicKey
-    {
+    pub fn get_prev_consensus_hashes<'a>(tx: &mut BurnDBTx<'a>, block_height: u64, first_block_height: u64, tip_block_hash: &BurnchainHeaderHash) -> Result<Vec<ConsensusHash>, db_error> {
         let mut i = 0;
         let mut prev_chs = vec![];
         while i < 64 && block_height - (((1 as u64) << i) - 1) >= first_block_height {
```

Comment:
I think loops like these are clearer and have less repetition if they iterate on blocks_back directly, rather than recomputing (2**i - 1), i.e.,

```
let mut cur_blocks_back = 0;
while cur_blocks_back < u64::max && block_height >= cur_blocks_back && block_height - cur_blocks_back >= first_block_height {
   let prev_block = block_height - cur_blocks_back;
   ...
   cur_blocks_back = (cur_blocks_back << 1) + 1
}
```

otherwise, the code ends up repeating the same calculation (left-shift and subtract) a bunch of times.

Reply:
Hmm, I have the opposite opinion -- I like seeing the full mathematical expression (even if it's repeated) because it makes the code easier to follow.  When evaluating how the code works in my mind, I don't have to remember how `cur_blocks_back` changes over multiple loop passes.  But, I'll happily make the change if this is the minority opinion.